File: python-ai-service/app/utils/file_utils.py
import os
import shutil
from pathlib import Path


def ensure_dir(path: str) -> None:
    """Create directory if it doesn't exist."""
    Path(path).mkdir(parents=True, exist_ok=True)


def safe_delete(path: str) -> None:
    """Delete a file, ignoring if it doesn't exist."""
    try:
        os.remove(path)
    except FileNotFoundError:
        pass


def get_file_size_mb(path: str) -> float:
    """Get file size in MB."""
    return os.path.getsize(path) / (1024 * 1024)

File: backend/src/utils/fileUtils.js
import fs from "fs/promises";
import path from "path";

/**
 * Ensure a directory exists, creating it recursively if needed.
 */
export async function ensureDir(dirPath) {
  await fs.mkdir(dirPath, { recursive: true });
}

/**
 * Safely delete a file, ignoring if it doesn't exist.
 */
export async function safeDelete(filePath) {
  try {
    await fs.unlink(filePath);
  } catch (err) {
    if (err.code !== "ENOENT") throw err;
  }
}

/**
 * Get file extension from a filename.
 */
export function getExtension(filename) {
  return path.extname(filename).toLowerCase();
}
